ci-operator/step-registry/baremetalds/assisted/operator/ztp/day2/baremetalds-assisted-operator-ztp-day2-commands.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+echo "************ baremetalds assisted operator ztp day2 command ************"
+
+# Fetch packet basic configuration
+# shellcheck source=/dev/null
+source "${SHARED_DIR}/packet-conf.sh"
+source "${SHARED_DIR}/ds-vars.conf"
+
+# Copy env variables set by baremetalds-devscripts-setup to remote server
+scp "${SSHOPTS[@]}" "${SHARED_DIR}/ds-vars.conf" "root@${IP}:ds-vars.conf"
+
+# Copy over the ENV variables set in remote_nodes.sh
+scp "${SSHOPTS[@]}" "${SHARED_DIR}/remote-vars.conf" "root@${IP}:remote-vars.conf"
+
+# ZTP scripts have a lot of default values for the spoke cluster configuration. Adding this so that they can be changed.
+if [[ -n "${ASSISTED_ZTP_CONFIG:-}" ]]; then
+  readarray -t config <<< "${ASSISTED_ZTP_CONFIG}"
+  for var in "${config[@]}"; do
+    if [[ ! -z "${var}" ]]; then
+      echo "export ${var}" >> "${SHARED_DIR}/assisted-ztp-config"
+    fi
+  done
+fi
+
+# Copy configuration for ZTP vars if present
+if [[ -e "${SHARED_DIR}/assisted-ztp-config" ]]
+then
+  scp "${SSHOPTS[@]}" "${SHARED_DIR}/assisted-ztp-config" "root@${IP}:assisted-ztp-config"
+fi
+
+tar -czf - . | ssh "${SSHOPTS[@]}" "root@${IP}" "cat > /root/assisted-service.tar.gz"
+
+# shellcheck disable=SC2087
+ssh "${SSHOPTS[@]}" "root@${IP}" bash - << EOF |& sed -e 's/.*auths\{0,1\}".*/*** PULL_SECRET ***/g'
+
+set -xeo pipefail
+
+REPO_DIR="/home/assisted-service"
+if [ ! -d "\${REPO_DIR}" ]; then
+  mkdir -p "\${REPO_DIR}"
+
+  echo "### Untar assisted-service code..."
+  tar -xzvf /root/assisted-service.tar.gz -C "\${REPO_DIR}"
+fi
+
+echo "### Adding remote nodes to spoke cluster..."
+
+# Inject variables from devscripts-remotenodes
+source /root/remote-vars.conf
+
+# Inject variables from devscripts-setup
+source /root/ds-vars.conf
+
+# TODO(lranjbar): Add CLUSTER_NAME and OCP_DIR to the devscripts-setup ds-vars.conf
+export OCP_DIR="\${DS_WORKING_DIR}/\${CLUSTER_NAME}"
+
+# Inject job configuration for ZTP, if available
+if [[ -e /root/assisted-ztp-config ]]
+then
+  source /root/assisted-ztp-config
+fi
+
+cd "\${REPO_DIR}/deploy/operator/ztp/"
+
+# TODO(lranjbar): Fix the permissions on this file
+#chmod 755 add_day2_remote_nodes.sh
+#./add_day2_remote_nodes.sh
+
+export REMOTE_BAREMETALHOSTS_FILE="${REMOTE_BAREMETALHOSTS_FILE:-/home/test/dev-scripts/ocp/ostest/remote_baremetalhosts.json}"
```

Comment:
Maybe I'm stating the obvious but, from looking at the logs, it seems like `REMOTE_BAREMETALHOSTS_FILE` is not set when the job gets here and, instead of `REMOTE_BAREMETALHOSTS_FILE` being `/opt/....` it ends up being `/home/test/dev-scripts/ocp/ostest/remote_baremetalhosts.json` which doesn't exist.

Reply:
Yes this is correct for this job run. This is the kind of stuff we end up debugging here. The way the values in these variables are set is what is usually the most problematic because it is complex. 

In our jobs we have two environments 1) the pod and 2) the remote server we ssh into. When you set a variable in a job configuration it sets the environment variable in the pod to equal that. However there can be a discrepancy for the environment variable on the remote SSH server. These variables get set every new pod (step) of the job as well.

Here is the official docs: https://docs.ci.openshift.org/docs/architecture/step-registry/#parameters We have added additional complexity with the remote server as well, but the propagation of the variables is complex too.

Reply:
This is a thing I've been thinking about a lot. Locally we pipe variables in from previous scripts a lot, the equivalent is steps here. It makes it even more confusing. Essentially in this run it didn't pipe in the right value from `devscripts-remotenodes` which runs before this. Most of the scripts we run locally have this awful piping in from random scripts and places making it hard to follow both locally and it's even worse in the CI.